### PR TITLE
[Forwardport] Add missing false-check to the ConfiguredRegularPrice price-model

### PR DESCRIPTION
**Pull request by mageprince**
### Original Pull Request

[#15129](https://github.com/magento/magento2/pull/15129)

### Description

`parent::getValue()` can return false while `$this->configuredOptions->getItemOptionsValue` only accepts float. So if the parent method returns false then it fails with the following error:

```
NOTICE: PHP message: PHP Fatal error:  Uncaught TypeError: Argument 1 passed to Magento\Catalog\Pricing\Price\ConfiguredOptions::getItemOptionsValue() must be of the type float, boolean given, called in /app/vendor/magento/module-catalog/Pricing/Price/ConfiguredRegularPrice.php on line 74 and defined in /app/vendor/magento/module-catalog/Pricing/Price/ConfiguredOptions.php:24
```

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @mageprince. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation

File: app/code/Magento/Catalog/Pricing/Price/ConfiguredRegularPrice.php [hunk 1]
```diff
@@ -13,12 +13,12 @@
 use Magento\Framework\Pricing\PriceCurrencyInterface;
 
 /**
- * Configured regular price model.
+ * Configured regular price model
```
Comment (orlangur) [inline app/code/Magento/Catalog/Pricing/Price/ConfiguredRegularPrice.php]: Restore all dots and don't touch them in the future.

File: app/code/Magento/Catalog/Pricing/Price/ConfiguredRegularPrice.php [hunk 2]
```diff
@@ -57,23 +57,22 @@ public function __construct(
      * @param ItemInterface $item
      * @return $this
      */
-    public function setItem(ItemInterface $item) : ConfiguredRegularPrice
+    public function setItem(ItemInterface $item)
```
Comment (orlangur) [inline app/code/Magento/Catalog/Pricing/Price/ConfiguredRegularPrice.php]: Do not remove existing type hints, feel free to add new ones.

**Review by orlangur — CHANGES_REQUESTED** [review]
@mageprince during forwardport original PR author must be preserved in commits.

File: app/code/Magento/Catalog/Pricing/Price/ConfiguredRegularPrice.php (hunk 2 above)
Comment (mageprince) [reply app/code/Magento/Catalog/Pricing/Price/ConfiguredRegularPrice.php]: Thanks for your review. I revert it back.

**Comment by mageprince** [comment]
Hi, @orlangur I have updated the files. Please check the PR.

**Comment by orlangur** [comment]
@mageprince please rewrite branch history so that it looks exactly as https://github.com/magento/magento2/pull/15129/commits (cherry-pick commits instead of redoing them)

**Comment by orlangur** [comment]
Port already done in https://github.com/magento/magento2/pull/16820

**Pull request closed without merging**